### PR TITLE
Update the code generator to use polyglot instead of old usage of polyglot-go

### PR DESCRIPTION
**Pull request by nicolaenicora**
## Description

1. Update the code generator to use polyglot instead of old usage of polyglot-go.
2. Add golint support; Fixed some linters errors

Fixes # (issue)

## Type of change

**Please update the title of this PR to reflect the type of change.** (Delete the ones that aren't required).

- [ x] Bug fix (non-breaking change which fixes an issue) [title: 'fix:']
- [ ] New feature (non-breaking change which adds functionality) [title: 'feat:']
- [ ] Refactor (non-breaking code changes that do not add new functionality or break existing functionality) [title: 'refactor:']
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please make sure that existing test cases pass (with `go test ./... -race` and `go test -bench=. ./... -race`),
and if required please add new test cases and list them below:

- [ ] Test A
- [ ] Test B

## Final Checklist:

- [x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,5 @@
 .env
 .idea/
 .DS_Store
-dist/
\ No newline at end of file
+dist/
+vendor
```
Comment (ShivanshVij) [inline .gitignore:7]: Good catch

File: .golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+linters:
```
Comment (ShivanshVij) [inline .golangci.yaml:1]: The linter should be added in a separate PR because we'll need to update the GH Actions as well

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.PHONY: all
```
Comment (ShivanshVij) [inline Makefile:1]: Same as above, this should be added in a different PR

File: pkg/generator/generator.go [hunk 4]
```diff
@@ -169,3 +156,24 @@ func (g *Generator) Generate(req *pluginpb.CodeGeneratorRequest) (res *pluginpb.
 
 	return plugin.Response(), nil
 }
+
+func extractNumberOfMethods(f *protogen.File) (int, int) {
```
Comment (ShivanshVij) [inline pkg/generator/generator.go:160]: Looks good

File: protoc-gen-go-frpc/main.go [hunk 5]
```diff
@@ -17,15 +17,16 @@
...
 	"os"
+
+	"github.com/loopholelabs/frpc-go/pkg/generator"
 )
 
 func main() {
 	gen := generator.New()
 
-	data, err := ioutil.ReadAll(os.Stdin)
+	data, err := io.ReadAll(os.Stdin)
```
Comment (ShivanshVij) [inline protoc-gen-go-frpc/main.go:29]: Good catch!

**Review by ShivanshVij — CHANGES_REQUESTED** [review]
The core changes look good, but there are some things that should probably be in a separate issue/in a separate PR. 

1. Please make sure you've signed off on the DCO (https://github.com/loopholelabs/frpc-go/pull/34/checks?check_run_id=16248886360). 

2. Remove the linter and makefile stuff from this PR, those should be added in a separate PR because we'll need to update the github actions to install and use those as well.

**Comment by nicolaenicora** [comment]
@ShivanshVij 
I will close this pull request and create 2 separate ones addressing them individually.

**Pull request closed without merging**